plugins/kinetic-dp/fu-kinetic-dp-secure-aux-isp.c

```diff
@@ -0,0 +1,1256 @@
+/*
+ * Copyright (C) 2021 Jeffrey Lin <jlin@kinet-ic.com>
+ * Copyright (C) 2022 Hai Su <hsu@kinet-ic.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "fu-kinetic-dp-secure-aux-isp.h"
+
+#include "fu-kinetic-dp-connection.h"
+
+struct _FuKineticDpSecureAuxIsp {
+	FuKineticDpAuxIsp parent_instance;
+};
+
+typedef struct {
+	guint32 isp_processed_size;
+	guint32 isp_total_size;
+	guint16 read_flash_prog_time;
+	guint16 flash_id;
+	guint16 flash_size;
+	gboolean is_isp_secure_auth_mode;
+} FuKineticDpSecureAuxIspPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(FuKineticDpSecureAuxIsp,
+			   fu_kinetic_dp_secure_aux_isp,
+			   FU_TYPE_KINETIC_DP_AUX_ISP)
```

Comment:
you don;t need to use G_DEFINE_TYPE_WITH_PRIVATE if you use G_DECLARE_FINAL_TYPE -- see the other plugins.